faiss/gpu/GpuIndex.h

```diff
@@ -23,6 +23,9 @@ struct GpuIndexConfig {
     /// On Pascal and above (CC 6+) architectures, allows GPUs to use
     /// more memory than is available on the GPU.
     MemorySpace memorySpace;
+
+    /// Should the index dispatch down to RAFT?
+    bool use_raft = false;
```

Comment:
Can you add a `#define USE_NVIDIA_RAFT` or similar in this file to enable/disable RAFT compilation and wrap all RAFT-using blocks with it?

Reply:
@wickedfoo, absolutely, I can do that. Just to clarify- you want to have `USE_NVIDIA_RAFT` instead of the `use_raft` member and not have both, right?

Reply:
As a follow-on question: Do we want to assume RAFT should always be enabled when `USE_NVIDIA_RAFT` is defined or do we want to keep something that is configurable at runtime?

Reply:
I would vote to keep it configurable at runtime, as it makes comparison easier. 
We could even *not* guard the use_raft field in the .h file and just fail at runtime if USE_NVIDIA_RAFT is not set at compile time. This pushes the test to the .cpp / .cu file.